packages/discovery-provider/src/challenges/play_count_milestones_challenge.py

```diff
@@ -0,0 +1,215 @@
+from datetime import datetime
+from typing import Dict, List, Optional, cast
+
+from sqlalchemy import and_, func
+from sqlalchemy.orm.session import Session
+
+from src.challenges.challenge import (
+    ChallengeManager,
+    ChallengeUpdater,
+    FullEventMetadata,
+)
+from src.challenges.challenge_event import ChallengeEvent
+from src.models.rewards.user_challenge import UserChallenge
+from src.models.social.play import Play
+from src.models.tracks.track import Track
+from src.utils.config import shared_config
+
+env = shared_config["discprov"]["env"]
+
+PLAY_MILESTONES = {250: "25", 1000: "100", 10000: "1000"}
+FINAL_MILESTONE = max(PLAY_MILESTONES.keys())
+
+if env == "stage" or env == "dev":
+    PLAY_MILESTONES = {1: "1", 2: "2", 3: "3"}
+    FINAL_MILESTONE = max(PLAY_MILESTONES.keys())
+
+
+class PlayCountMilestonesUpdater(ChallengeUpdater):
+    """
+    Challenge to reward users for achieving specific play count milestones in 2025.
+
+    The milestones are:
+    - 250 plays: 25 AUDIO
+    - 1000 plays: 100 AUDIO
+    - 10000 plays: 1000 AUDIO
+
+    Each milestone is a separate challenge that is marked as complete when reached.
+    """
+
+    def get_user_play_count_2025(self, session: Session, user_id: int) -> int:
+        """Get the total play count for an artist's tracks in 2025"""
+        start_date = datetime(2025, 1, 1)
+        end_date = datetime(2026, 1, 1)
+
+        play_count = (
+            session.query(func.count(Play.id))
+            .join(
+                Track,
+                and_(
+                    Track.track_id == Play.play_item_id,
+                    Track.owner_id == user_id,
+                    Track.is_current == True,
+                    Track.is_delete == False,
+                ),
+            )
+            .filter(and_(Play.created_at >= start_date, Play.created_at < end_date))
+            .scalar()
+        )
+
+        return cast(int, play_count)
+
+    def _get_completed_milestones(self, session: Session, user_id: int) -> List[int]:
```

Comment:
seems to me like we could just return the max completed_milestone in this fn instead of an array of all the completed milestones?